pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -422,54 +423,59 @@ func (f *framework) RunUnreservePlugins(
 func (f *framework) RunPermitPlugins(
 	pc *PluginContext, pod *v1.Pod, nodeName string) *Status {
 	timeout := maxTimeout
-	statusCode := Success
+	var needsWait bool
 	for _, pl := range f.permitPlugins {
 		status, d := pl.Permit(pc, pod, nodeName)
-		if !status.IsSuccess() {
-			if status.Code() == Unschedulable {
-				msg := fmt.Sprintf("rejected by %v at permit: %v", pl.Name(), status.Message())
-				klog.V(4).Infof(msg)
-				return NewStatus(status.Code(), msg)
-			}
-			if status.Code() == Wait {
-				// Use the minimum timeout duration.
-				if timeout > d {
-					timeout = d
-				}
-				statusCode = Wait
-			} else {
-				msg := fmt.Sprintf("error while running %v permit plugin for pod %v: %v", pl.Name(), pod.Name, status.Message())
-				klog.Error(msg)
-				return NewStatus(Error, msg)
+		if status.IsSuccess() {
+			continue
+		}
+
+		switch status.Code() {
+		case Unschedulable:
+			msg := fmt.Sprintf("rejected by %v at permit: %v", pl.Name(), status.Message())
+			klog.V(4).Infof(msg)
+			return NewStatus(status.Code(), msg)
+		case Wait:
+			if timeout > d {
+				timeout = d
 			}
+			needsWait = true
+		default:
+			msg := fmt.Sprintf("error while running %v permit plugin for pod %v: %v", pl.Name(), pod.Name, status.Message())
+			klog.Error(msg)
+			return NewStatus(Error, msg)
 		}
 	}
 
+	if !needsWait {
+		return nil
+	}
+
 	// We now wait for the minimum duration if at least one plugin asked to
 	// wait (and no plugin rejected the pod)
-	if statusCode == Wait {
-		w := newWaitingPod(pod)
-		f.waitingPods.add(w)
-		defer f.waitingPods.remove(pod.UID)
-		timer := time.NewTimer(timeout)
-		klog.V(4).Infof("waiting for %v for pod %v at permit", timeout, pod.Name)
-		select {
-		case <-timer.C:
-			msg := fmt.Sprintf("pod %v rejected due to timeout after waiting %v at permit", pod.Name, timeout)
+	w := newWaitingPod(pod)
+	f.waitingPods.add(w)
+	defer f.waitingPods.remove(pod.UID)
+	timer := time.NewTimer(timeout)
+	klog.V(4).Infof("waiting for %v for pod %v at permit", timeout, pod.Name)
```

Comment:
It would be useful to also log the plugin name.

Reply:
It could be a list of plugins since more than one could return a wait timeout